web/modules/custom/joinup_communities/tallin/src/Plugin/Field/FieldWidget/TallinEntryWidget.php

```diff
@@ -0,0 +1,171 @@
+ * a link field and an option select field. Each of them will be constructed,
+ * validated and contain widget similar or the same as their individual plugins.
+ *
+ * @see \Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextareaWidget
+ * @see \Drupal\Core\Field\Plugin\Field\FieldWidget\OptionsSelectWidget
+ * @see \Drupal\link\Plugin\Field\FieldWidget\LinkWidget
+ *
+ * @FieldWidget(
+ *   id = "tallin_entry_default",
+ *   label = @Translation("Tallin entry widget"),
+ *   field_types = {
+ *     "tallin_entry"
+ *   }
+ * )
+ */
+class TallinEntryWidget extends WidgetBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a SearchWidget object.
+   *
+   * @param string $plugin_id
+   *   The plugin_id for the widget.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the widget is associated.
+   * @param array $settings
+   *   The widget settings.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $item = $items[$delta];
+
+    $element['wrapper'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->fieldDefinition->getLabel(),
+    ];
+
+    $element['wrapper']['tallin_description'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'h3',
+      '#value' => $this->fieldDefinition->getSetting('tallin_description'),
+      '#weight' => 0,
+    ];
+
+    $element['wrapper']['explanation'] = [
+      '#type' => 'text_format',
+      '#title' => t('Explanations'),
+      '#default_value' => $item->value,
+      '#format' => $item->format,
+      '#weight' => 1,
+    ];
+
+    $element['wrapper']['status'] = [
```

Comment:
Maybe the status should be on top of the explanation field, because the latter is expected to be an explanation of the former?